astropy/units/physical.py

```diff
@@ -485,19 +485,16 @@ def def_physical_type(unit: core.UnitBase, name: str | set[str]) -> None:
     unit_already_in_use = physical_type_id in _physical_unit_mapping
     if unit_already_in_use:
         physical_type = _physical_unit_mapping[physical_type_id]
-        physical_type_names |= set(physical_type)
-        physical_type.__init__(unit, physical_type_names)
+        physical_type._physical_type = sorted(physical_type_names | set(physical_type))
     else:
         physical_type = PhysicalType(unit, physical_type_names)
         _physical_unit_mapping[physical_type_id] = physical_type
 
     for ptype in physical_type:
         _unit_physical_mapping[ptype] = physical_type_id
-
-    for ptype_name in physical_type_names:
-        _name_physical_mapping[ptype_name] = physical_type
+        _name_physical_mapping[ptype] = physical_type
```

Comment:
Had to think a bit, but the logic here is that the names are also normalized in `PhysicalType` initialization, so this is always good, nice.